vcd/resource_vcd_catalog.go

```diff
@@ -81,13 +84,14 @@ func resourceVcdCatalogRead(d *schema.ResourceData, meta interface{}) error {
 		return fmt.Errorf(errorRetrievingOrg, err)
 	}
 
-	catalog, err := adminOrg.FindCatalog(d.Id())
-	if err != nil || catalog == (govcd.Catalog{}) {
+	catalog, err := adminOrg.GetCatalogByNameOrId(d.Id(), false)
+	if err != nil || catalog == nil {
 		log.Printf("[DEBUG] Unable to find catalog. Removing from tfstate")
 		d.SetId("")
-		return nil
+		return fmt.Errorf("error retrieving catalog %s : %s", d.Id(), err)
 	}
 
+	d.SetId(catalog.Catalog.ID)
```

Comment:
We should add message in chagelog that ID in state now real one against old one - name.

Reply:
done